### ISSUE TITLE
unexpected spaces in scientific notation

### ISSUE THREAD

**Issue by YuxiLing** [PRQL/prql#3188]
### What happened?

It is similar to #1118 and #2860 (but not fixed after #2865). But I found this issue in the newest version.

PRQL compiler version:0.9.2

I found that prql compiler converts the big number (greater than 1e19 or approximately 2^63) to the format with scientifc notation. However, a space is in generated sql.

Error message from db: OperationalError: near "AS": syntax error

### PRQL input

```elm
from tb1
select new_col=c0+ 10000000000000000000
```

### SQL output

```SQL
SELECT
  c0 + 1 e19 AS new_col
FROM
  tb1
```

### Expected SQL output

```SQL
SELECT
  c0 + 1E19 AS new_col
FROM
  tb1
```

### MVCE confirmation

- [X] Minimal example
- [X] New issue

**Comment by max-sixty** [comment]
Thanks @YuxiLing , I added this to https://github.com/shssoichiro/sqlformat-rs/issues/22.
